src/App.tsx

```diff
@@ -15,47 +16,135 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType !== SortType.NONE) {
+    visibleGoods.sort((firstGood, secondGood) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return firstGood.localeCompare(secondGood);
+        case SortType.LENGTH:
+          return firstGood.length - secondGood.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const goods = getReorderedGoods(goodsFromServer, { isReversed, sortType });
+
+  const handleSortChange = (newSortType: SortType) => {
+    setSortType(newSortType);
+  };
+
+  const handleReverseChange = () => {
+    setIsReversed(previous => !previous);
+  };
+
+  const resetState = () => {
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button',
+            'is-info',
+            {
+              'is-light': sortType !== SortType.ALPHABET,
+            },
+          )}
+          onClick={() => {
+            handleSortChange(SortType.ALPHABET);
+          }}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames(
+            'button',
+            'is-info',
```

Comment:
four buttons === four different colors 
```suggestion
            'is-success',
```

Reply:
<img width="568" alt="Screenshot 2023-03-19 at 16 55 26" src="https://user-images.githubusercontent.com/91826635/226184302-31d357cb-5c7a-4bd4-92d4-b0968e603aef.png">